From 495a230257cf936cadbefdfc88e416d120b7ce68 Mon Sep 17 00:00:00 2001
From: Daniel Scherzer <daniel@wikiteq.com>
Date: Thu, 25 May 2023 13:24:49 +0200
Subject: [PATCH] Hooks: avoid premature access to MediaWikiServices

Rather than retrieving configuration from MediaWikiServices, use the
underlying global variables, since when extension functions are
called the services are not ready yet, and accessing them is
deprecated.
---
 EmbedVideo.hooks.php | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/EmbedVideo.hooks.php b/EmbedVideo.hooks.php
index 19327dc..6af611b 100644
--- a/EmbedVideo.hooks.php
+++ b/EmbedVideo.hooks.php
@@ -1,7 +1,5 @@
 <?php
 
-use MediaWiki\MediaWikiServices;
-
 /**
  * EmbedVideo
  * EmbedVideo Hooks
@@ -74,14 +72,15 @@ class EmbedVideoHooks {
 	public static function onExtension() {
 		global $wgEmbedVideoDefaultWidth, $wgMediaHandlers, $wgFileExtensions;
 
-		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig('main');
+		global $wgEmbedVideoEnableAudioHandler, $wgEmbedVideoEnableVideoHandler,
+				$wgEmbedVideoAddFileExtensions;
 
 		if (!isset($wgEmbedVideoDefaultWidth) && (isset($_SERVER['HTTP_X_MOBILE']) && $_SERVER['HTTP_X_MOBILE'] == 'true') && $_COOKIE['stopMobileRedirect'] != 1) {
 			// Set a smaller default width when in mobile view.
 			$wgEmbedVideoDefaultWidth = 320;
 		}
 
-		if ($config->get('EmbedVideoEnableAudioHandler')) {
+		if ($wgEmbedVideoEnableAudioHandler) {
 			$wgMediaHandlers['application/ogg']		= 'EmbedVideo\AudioHandler';
 			$wgMediaHandlers['audio/flac']			= 'EmbedVideo\AudioHandler';
 			$wgMediaHandlers['audio/ogg']			= 'EmbedVideo\AudioHandler';
@@ -91,7 +90,7 @@ class EmbedVideoHooks {
 			$wgMediaHandlers['audio/webm']			= 'EmbedVideo\AudioHandler';
 			$wgMediaHandlers['audio/x-flac']		= 'EmbedVideo\AudioHandler';
 		}
-		if ($config->get('EmbedVideoEnableVideoHandler')) {
+		if ($wgEmbedVideoEnableVideoHandler) {
 			$wgMediaHandlers['video/mp4']			= 'EmbedVideo\VideoHandler';
 			$wgMediaHandlers['video/ogg']			= 'EmbedVideo\VideoHandler';
 			$wgMediaHandlers['video/quicktime']		= 'EmbedVideo\VideoHandler';
@@ -99,7 +98,7 @@ class EmbedVideoHooks {
 			$wgMediaHandlers['video/x-matroska']	= 'EmbedVideo\VideoHandler';
 		}
 
-		if ($config->get('EmbedVideoAddFileExtensions')) {
+		if ($wgEmbedVideoAddFileExtensions) {
 			$wgFileExtensions[] = 'flac';
 			$wgFileExtensions[] = 'mkv';
 			$wgFileExtensions[] = 'mov';
-- 
2.28.0.windows.1

